src/jit/decomposelongs.cpp

```diff
@@ -1013,15 +1013,25 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 {
     assert(use.IsInitialized());
 
-    GenTree* tree      = use.Def();
-    GenTree* gtLong    = tree->gtGetOp1();
+    GenTree* shift     = use.Def();
```

Comment:
Hurray for using descriptive variable names! But this should have been done separately, a lot of changes in this diff are caused by this and not by the actual bug fix.